### PR TITLE
Bitmex fetch funding rate history

### PR DESCRIPTION
**Pull request by samgermain**
There's a lot of symbols in the funding rate history that aren't in the markets, so they don't get converted, for example `ETHXBT`, which I think might be referring to `.BETHXBT`. I'm not sure what the best way to handle this problem would be.

------------------------------

```
2022-05-16T22:54:04.823Z                                                                   2022-05-16T23:00:03.135Z                                                            2022-05-16T23:02:27.262Z                                                         2022-05-16T23:03:52.462Z                                                       2022-05-16T23:04:41.677Z                                                            2022-05-16T23:05:13.793Z
Node.js: v14.17.0                                                                          Node.js: v14.17.0                                                                   Node.js: v14.17.0                                                                Node.js: v14.17.0                                                              Node.js: v14.17.0                                                                   Node.js: v14.17.0
CCXT v1.82.39                                                                              CCXT v1.82.62                                                                       CCXT v1.82.62                                                                    CCXT v1.82.62                                                                  CCXT v1.82.62                                                                       CCXT v1.82.62
bitmex.fetchFundingRateHistory ()                                                          bitmex.fetchFundingRateHistory (BTC/USD:BTC)                                        bitmex.fetchFundingRateHistory (, 1648771200000)                                 bitmex.fetchFundingRateHistory (, , 3)                                         bitmex.fetchFundingRateHistory (BTC)                                                bitmex.fetchFundingRateHistory (BTC:perpetual)
2022-05-16T22:54:08.490Z iteration 0 passed in 2861 ms                                     2022-05-16T23:00:06.099Z iteration 0 passed in 2249 ms                              2022-05-16T23:02:29.883Z iteration 0 passed in 2201 ms                           2022-05-16T23:03:55.454Z iteration 0 passed in 2195 ms                         2022-05-16T23:04:44.372Z iteration 0 passed in 2233 ms                              2022-05-16T23:05:20.293Z iteration 0 passed in 5813 ms
     symbol |              fundingRate |     timestamp |                 datetime               symbol |             fundingRate |     timestamp |                 datetime                symbol | fundingRate |     timestamp |                 datetime                                                                                          symbol |             fundingRate |     timestamp |                 datetime         symbol |             fundingRate |     timestamp |                 datetime
---------------------------------------------------------------------------------          --------------------------------------------------------------------------------    ---------------------------------------------------------------------------      symbol |            fundingRate |     timestamp |                 datetime     --------------------------------------------------------------------------------    --------------------------------------------------------------------------------
     ETHXBT       |    0.0010890000000000001 | 1462622400000 | 2016-05-07T12:00:00.000Z    BTC/USD:BTC |   0.0006900000000000001 | 1463227200000 | 2016-05-14T12:00:00.000Z           BTC/USD:BTC |      0.0001 | 1648785600000 | 2022-04-01T04:00:00.000Z      --------------------------------------------------------------------------     BTC/USD:BTC |   0.0006900000000000001 | 1463227200000 | 2016-05-14T12:00:00.000Z    BTC/USD:BTC |   0.0006900000000000001 | 1463227200000 | 2016-05-14T12:00:00.000Z
...                                                                                        ...                                                                                 ...                                                                              ETHXBT |  0.0010890000000000001 | 1462622400000 | 2016-05-07T12:00:00.000Z     ...                                                                                 ...
     BTC/USD:BTC  |    0.0009739999999999999 | 1464868800000 | 2016-06-02T12:00:00.000Z    BTC/USD:BTC |                0.001416 | 1467259200000 | 2016-06-30T04:00:00.000Z           AXS/USD:BTC |      0.0001 | 1648843200000 | 2022-04-01T20:00:00.000Z      ETHXBT |               0.000328 | 1462708800000 | 2016-05-08T12:00:00.000Z     BTC/USD:BTC |                0.001416 | 1467259200000 | 2016-06-30T04:00:00.000Z    BTC/USD:BTC |                0.001416 | 1467259200000 | 2016-06-30T04:00:00.000Z
     LSKXBT       |                 0.000235 | 1464955200000 | 2016-06-03T12:00:00.000Z    BTC/USD:BTC |                0.002604 | 1467288000000 | 2016-06-30T12:00:00.000Z           EOS/USD:BTC |      0.0001 | 1648843200000 | 2022-04-01T20:00:00.000Z      ETHXBT | 0.00022900000000000004 | 1462795200000 | 2016-05-09T12:00:00.000Z     BTC/USD:BTC |                0.002604 | 1467288000000 | 2016-06-30T12:00:00.000Z    BTC/USD:BTC |                0.002604 | 1467288000000 | 2016-06-30T12:00:00.000Z
100 objects                                                                                100 objects                                                                         100 objects                                                                      3 objects                                                                      100 objects                                                                         100 objects
```

### CODE REVIEW

File: js/bitmex.js [hunk 1]
```diff
@@ -1746,6 +1748,88 @@ module.exports = class bitmex extends Exchange {
...
+         * @param {bool} params.reverse if true, will sort results newest first
+         * @param {int} params.start starting point for results
+         * @param {str} params.columns array of column names to fetch in info, if omitted, will return all columns
+         * @param {str} params.filter generic table filter, send json key/value pairs, such as {"key": "value"}, you can key on individual fields, and do more advanced querying on timestamps, see the [timestamp docs]{@link https://www.bitmex.com/app/restAPI#Timestamp-Filters} for more details
+         * @returns A list of [funding rate history structures]{@link https://docs.ccxt.com/en/latest/manual.html#funding-rate-history-structure}
+         */
+        await this.loadMarkets ();
+        const request = {};
+        let market = undefined;
+        if (symbol in this.currencies) {
```
Comment (kroitor) [inline js/bitmex.js:1770]: I'm not sure if it's legit to lookup a trading pair (a symbol) in a dictionary of currencies in this case. Shouldn't we lookup a market and take its base or quote currency instead?

Comment (samgermain) [reply js/bitmex.js:1770]: The api accepts both, so if we don't accept both, then we're limiting the api a bit

**Comment by samgermain** [comment]
@kroitor There was still an issue with this PR, in my top comment

**Comment by kroitor** [comment]
> @kroitor There was still an issue with this PR, in my top comment
> 
> If I have any text in my top comment other than the tests, it's usually a question

I will look into it.

**Comment by kroitor** [comment]
@samgermain what is the issue exactly? Is it about the symbols it could not parse? What can be done with that theoretically ?

**Comment by samgermain** [comment]
> @samgermain what is the issue exactly? Is it about the symbols it could not parse? What can be done with that theoretically ?

I don't really know what to do, it looks like they only reference an index symbol, I'm not exactly sure why they would show up here but not in fetch markets

**Pull request merged**